airsenal/scripts/replay_season.py

```diff
@@ -72,21 +79,41 @@ def replay_season(
                 tag_prefix=tag_prefix,
                 dbsession=session,
             )
+        gw_result = {"gameweek": gw, "predictions_tag": tag}
+
         if not transfers:
             continue
         if gw == gameweek_start and new_squad:
             print("Creating initial squad...")
-            fill_initial_squad(tag, gw_range, season, fpl_team_id)
+            squad = fill_initial_squad(tag, gw_range, season, fpl_team_id)
         else:
             print("Optimising transfers...")
-            run_optimization(
+            squad = run_optimization(
                 gw_range,
                 tag,
                 season=season,
                 fpl_team_id=fpl_team_id,
                 num_thread=num_thread,
             )
+        gw_result["starting_11"] = []
+        gw_result["subs"] = []
+        for p in squad.players:
+            if p.is_starting:
+                gw_result["starting_11"].append(p.name)
+            else:
+                gw_result["subs"].append(p.name)
+            if p.is_captain:
+                gw_result["captain"] = p.name
+            elif p.is_vice_captain:
+                gw_result["vice_captain"] = p.name
+        exp_points = squad.get_expected_points(gw, tag)
+        gw_result["expected_points"] = exp_points
+        actual_points = squad.get_actual_points(gw, season)
+        gw_result["actual_points"] = actual_points
```

Comment:
As well as not being sure if the hits are being taken account, I'm not sure if bench boost weeks are being counted properly. When we call `run_optimization`, can we check the number of hits taken and if bench boost was used for the next week?

Reply:
Just spoke to Jack about this - so I see currently the replay functionality doesn't consider chips right now, so we can ignore the bench boost / t-c stuff for now. But I think I can try to fix the bit about consider points hits - will work on this tomorrow hopefully! (Just need a few fixes to what `run_optimizer` returns - we think this needs to return the best_strategy which should contain how many hits were taken at each week)